app/models/message.rb

```diff
@@ -59,4 +55,12 @@ def chat_message_link
       anchor: "chat-row-#{id}"
     )
   end
+
+  private
+
+  def send_outbound_messages
+    [PostmarkAdapter::Outbound, SignalAdapter::Outbound, TelegramAdapter::Outbound, ThreemaAdapter::Outbound].each do |adapter|
+      adapter.send!(self)
+    end
+  end
```

Comment:
I have two suggestions:

1) Could we move the condition from `after_create_commit` into the method (as a guard clause). This ensures that no matter how this method is called (i.e., not by a lifecycle hook), it will only send outbound messages. Also, I’d prefer to have all the logic in one place.

2) Can we rename this method to `send_if_outbound` or something similar? I think I see where the current name comes from (with “messages” referring to the actual email or Signal/Threema/Telegram message instead of the `Message` record in our database). However, we already use the term “message” for our `Message` model, so I’d prefer if we didn’t use it for two things with slightly different meanings.

Reply:
@mattwr18 Just saw that you changed that literally minutes ago!

Reply:
Yes, I also just commented on why I changed it. It was pretty similar to the reason you gave, except for with the caveat that it was buggy with the inline conditional/proc and a test was failing. Thanks for tests!

2. I don't know if I understand your concern. It is referring to the `Message` model as this is what we send with our adapters. The Outbound adapters accept `self` as a param, which is the instance of `Message`, they then call `perform_later` with `message` and/or `message.text`

Reply:
> I guess send_if_outbound doesn't seem to address the manually_created? part of the condition.

A message that’s created manually is always an inbound message.

Reply:
> so the method is actually responsible for sending out messages.

Yep, and that’s the point. I’d prefer if we didn’t use the term *Message* for both the `Message` record type and the Signal/Threema/Telegram/email message. Even though they are obviously related, they are different and it introduces ambiguity.

I’d like to think of this as "sending out a single message via multiple channels".

(Also in practice, we never send out a message via multiple channels because contributors onboard using a single channel so the fact that we loop over multiple adapters is more of an implementation detail.)

Reply:
> I’d prefer if we didn’t use the term Message for both the Message record type and the Signal/Threema/Telegram/email message.

I am not using it to refer to the Signal/Threema/Telegram/email message, I am using it to refer to the instance of the `Message` class, which is being sent by the channel.

> and it introduces ambiguity.

It doesn't for me cause it's clear to me that message refers to the instance of the `Message` class that is being sent. Also, `Telegram` for example already uses `send_message` as the name of their method so the ambiguity, if there is any, already exists to some degree.

All this being said, I don't have an issue with your proposed naming, I just want to make sure that we are not asking for changes to code for personal code preferences, which we all have and are different from each other sometimes.